src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/VsContainedLanguageComponentsFactory.cs

```diff
@@ -63,14 +65,18 @@ protected override Task DisposeCoreAsync(bool initialized)
             IVsHierarchy myHierarchy = null;
             IVsContainedLanguageFactory myContainedLanguageFactory = null;
 
-            ProjectServices.ThreadingService.JoinableTaskFactory.Run(async () =>
+            _projectVsServices.ThreadingService.JoinableTaskFactory.Run(async () =>
             {
                 await InitializeAsync().ConfigureAwait(false);
 
-                await ProjectServices.ThreadingService.JoinableTaskFactory.SwitchToMainThreadAsync();
+                Guid? languageServiceId = await GetLanguageServiceId().ConfigureAwait(false);
```

Comment:
This is a CLSID, throughout VS it's referred to as "Language Service ID". GUID describes the data format, not what it is.